mlx_embeddings/models/xlm_roberta.py

```diff
@@ -339,9 +342,92 @@ def __call__(
         head_mask = self.get_head_mask(head_mask, self.config.num_hidden_layers)
 
         embedding_output = self.embeddings(input_ids, token_type_ids, position_ids)
-        encoder_outputs = self.encoder(embedding_output, extended_attention_mask)
+        encoder_outputs = self.encoder(
+            embedding_output, 
+            extended_attention_mask,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            )
         sequence_output = encoder_outputs[0]
         pooled_output = (
             self.pooler(sequence_output) if self.pooler is not None else None
         )
-        return (sequence_output, pooled_output) + encoder_outputs[1:]
+
+        if not return_dict:
+            return (sequence_output, pooled_output) + encoder_outputs[1:]
+        
+        return {
+            "embeddings": pooled_output,
+            "last_hidden_state": sequence_output,
+            "hidden_states": encoder_outputs[1] if output_hidden_states else None,
+            "attentions": encoder_outputs[2] if output_attentions else None,
+        }
```

Comment:
Check if not return dict vs your new return.

Also. pooled_output can be none.